packages/common/src/activity-options.ts

```diff
@@ -100,6 +101,14 @@ export interface ActivityOptions {
    * @default true
    */
   allowEagerDispatch?: boolean;
+
+  /**
+   * When using the Worker Versioning feature, specifies whether this Activity should run on a
+   * worker with a compatible Build Id or not. See {@link VersioningIntent}.
+   *
+   * @experimental
+   */
+  versioningIntent?: VersioningIntent;
```

Comment:
This is a bit messy but for _most_ of the SDK we used enums for this type of option: https://github.com/temporalio/sdk-typescript/blob/ab58f964afe2c0ffb22107b1802a656e35cdc7b6/packages/common/src/activity-options.ts#L9

Reply:
I would do it the way you did it here if I had to do it all over again

Reply:
So do you want it to be an enum? I dunno that being consistent like that is terribly important, if this is slightly nicer.

Reply:
I'm generally in favor of consistency, yet I really don't mind going against the trend on this precise question. TypeScript enums unfortunately have quite a few quirks, so I'd be happy to walk away from them from now on. Plus we already have a few places where we used unions of strings for things that could have been enums instead.

My only concern here is that you made those strings Titlecase. In most places where we already used unions of strings, we used ALL_UPPERCASE syntax, with a very few places using camelCase. I think I'd prefer staying with ALL_UPPERCASE, as it makes it more obvious when looking at TS code that it represents some constant keyword, where other syntaxes tends to look like just plain strings.

> I would do it the way you did it here if I had to do it all over again

Not a priority, but we could easily consider accepting both strings and enums in all of these places, and deprecate these enums, so people slowly move away from them. It might be a little bit harder to get away from cases where SDK gives back such enums, but actually, I don't think think there are that much cases.

Reply:
I like the suggestion of transitioning to string unions everywhere. I think our string union types are all camel case though. @mjameswh can you come up with a guideline?

Reply:
@bergundy I just did some search on the code.

Except for string union types introduced by the Versioning feature (which is still experimental and probably almost unused in the wild yet), all publicly exported string unions are ALL_UPPERCASE.

The only places where we are not doing so are types that are union of other types/interfaces with a discriminant field (for example `EphemeralServerExecutable.type` can be either `'cached-download'` or `'existing-path'; similar for `HeartbeatInput`.

Existing enums were also very consistently using ALL_UPPERCASE.

So I'd maintain that convention.

Reply:
Schedule action is camel case..

Reply:
> Schedule action is camel case..

Yeah. No convention holds at the moment for discriminants in union of types/interfaces. I did not fixed these in Versioning either, as I was running out of time to cut a release today.

I will continue analyze on this Monday.